carmin.yaml

```diff
@@ -266,6 +265,133 @@ paths:
           description: successful response
         default:
           $ref: '#/responses/GenericError'
+  /path/{uri}:
+    get:
+      description: Shows the attributes of a given path
+      operationId: getPath
+      parameters:
+        - name: uri
+          in: path
+          required: true
+          description: path URI
+          type: string
+          format: ascii
+      responses:
+        '200':
+          description: successful response
+          schema:
+            $ref: '#/definitions/Path'
+        default: 
+          $ref: '#/responses/GenericError' 
+  /path/exists/{uri}:
+    get:
+      description: Tests if the path exists
+      operationId: existsPath
+      parameters:
+        - name: uri
+          in: path
+          required: true
+          description: path URI
+          type: string
+          format: ascii
+      responses:
+        '200':
+          description: successful response
+          schema:
+            type: boolean
+        default: 
+          $ref: '#/responses/GenericError'       
+  /path/{uri}/delete:
+    delete:
+      description: Deletes the path. Returns an error if user is not allowed.
+      operationId: deletePath
+      parameters:
+        - name: uri
+          in: path
+          required: true
+          description: path URI
+          type: string
+          format: ascii
+      responses:
+        '204':
+          description: successful response
+        default: 
+          $ref: '#/responses/GenericError'
+  /path/{uri}/list:
+    get:
+      description: When uri refers to a directory, returns a list of Paths in this directory.
+      operationId: listDirectory
+      parameters:
+        - name: uri
+          in: path
+          required: true
+          description: path URI
+          type: string
+          format: ascii
+      responses:
+        '200':
+          description: successful response
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Path'
+        default: 
+          $ref: '#/responses/GenericError'
+  /path/{uri}/mkdir:
+    get:
+      description: Creates a directory at uri and returns the corresponding Path object.
+      operationId: createDirectory
+      parameters:
+        - name: uri
+          in: path
+          required: true
+          description: path URI
+          type: string
+          format: ascii
+      responses:
+        '200':
+          description: successful response
+          schema:
+              $ref: '#/definitions/Path'
+        default: 
+          $ref: '#/responses/GenericError'
+  /path/{uri}/content:
```

Comment:
Why not have methods that return the raw files ? The content would be lighter (no base64) and the client would not have to decode it.

Reply:
you mean we should revise the spec to avoid base64 alltogether?

Reply:
I don't know why base64 was choosed in the first place but I can't see a good reason to use exclusively it.
The advantage of base64 is I think to be textual and to be included in a json response.
I thought of a mixed solution : 
- rest methods to download/upload raw files.
- a "base64content" property in the path object that would be filled in the getPath request response when an argument "withContent" is present

Reply:
ok, let's discuss it on Sept 6.